public/assets/js/dropzoneFunctions.js

```diff
@@ -1,53 +1,110 @@
-
-function drop_handler(event) {
-    event.preventDefault();
-    // if dropped items aren't files, reject them
-    var dt = event.dataTransfer;
-    if (dt.items) {
-        if (dt.items[0].kind == 'file') {
-            var droppedFile = dt.items[0].getAsFile();
-            publishFileFunctions.previewAndStageFile(droppedFile);
+const dropzoneFunctions = {
+    triggerFileChooser: function (fileInputId) {
+        document.getElementById(fileInputId).click();
+    },
+    dragover_handler: function (event) {
+        event.preventDefault();
+    },
+    dragend_handler: function (event) {
+        var dt = event.dataTransfer;
+        if (dt.items) {
+            for (var i = 0; i < dt.items.length; i++) {
+                dt.items.remove(i);
+            }
+        } else {
+            event.dataTransfer.clearData();
         }
-    }
-}
+    },
+    primary: {
+        drop_handler: function (event) {
+            event.preventDefault();
+            this.dragexit_handler();
+            var dt = event.dataTransfer;
+            if (dt.items) {
+                if (dt.items[0].kind == 'file') {
```

Comment:
If I drop something that is not a file, should I receive feedback on that? I'm not actually sure what you could drop that _isn't_ a file, so I'm not sure whether or not someone would be confused when they did so.

Reply:
tbh I got the pattern from mdn and wasn't sure: https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API/File_drag_and_drop.
I looked into it, and other elements on the page are one item that can be dragged and dropped that isn't a file.  e.g. you can drag one of the navigation elements over the drop zone.  
I will add a catch for that so it updates the error message accordingly.